spec/predicates/human-review.md

```diff
@@ -0,0 +1,90 @@
+# Predicate type: Human Reviews
+
+Type URI: https://in-toto.io/attestation/human-review/v0.1
+
+Version: 0.1.0
+
+Authors:
+
+## Purpose
+
+This is a generic predicate type used to describe human reviews of software
+artifacts. For example, this predicate (or a derivative) can be used to wrap
+results of code reviews, software audits, and dependency reviews such as the
+crev project.
+
+## Use Cases
+
+Software supply chains encompass many types of human reviews. Best practices
+including standards like SLSA recommend two or multi-party review for source
+code. Another example is the crev project where open source software
+dependencies are socially reviewed.
+
+## Prerequisites
+
+Understanding of the
+[in-toto attestation specification](https://github.com/in-toto/attestation).
+
+## Model
+
+This predicate type includes one compulsory field, `result`, that describes the
+result of the review. Derivatives of this predicate may include other contextual
+fields pertaining to different code review systems or use cases. The predicate
+includes an optional `reviewLink` field that can be used to find the full
+review. In the general case, the subject of a code review attestation must
+identify the software artifacts the review applies to.
+
+## Schema
+
+```json
+{
+    "_type": "https://in-toto.io/Statement/v1",
+    "subject": [{...}],
+    "predicateType": "https://in-toto.io/attestation/human-review/v0.1",
+    "predicate": {
+        "result": "pass|fail",
+        "reviewLink": "<URL>",
+        "timestamp": "<TIMESTAMP>"
+    }
+}
+```
+
+### Parsing Rules
+
+This predicate follows the
+[in-toto Attestation Framework's parsing rules](../v1/README.md#parsing-rules).
+
+### Fields
+
+`result` _enum_, _optional_
+
+Indicates the result of the review. For example, it may have the values "pass"
+or "fail".
+
+`reviewLink` _URI_, _optional_
+
+Contains a link to the full review. Useful to point to information that cannot
+be captured in the attestation.
+
+`timestamp` _Timestamp_, _optional_
+
+Indicates time of review creation.
+
+## Example
+
+```json
+{
+    "_type": "https://in-toto.io/Statement/v1",
+    "subject": [{
+        "digest": {
+            "gitCommit": "b0bd8ecab5607e174fa403e002c74a666e7edd51"
+        }
+    }],
+    "predicateType": "https://in-toto.io/attestation/human-review/v0.1",
+    "predicate": {
+        "result": "pass",
```

Comment:
Maybe I'm missing something: How come there is no `reviewer` field in this predicate? I also would have expected to see a `result` field in the VCS predicate, hence my earlier comment about the two predicates being practically identical.

Reply:
This one is a more generic review type. Here, the reviewer is expected to be identified using the signature rather than a field in the predicate. The vcs type is to support, say, github or a gitlab server issuing an attestation with an online key.

Reply:
The `result` field being omitted in the VCS one is because in the VCS case, we don't want negative reviews. An attestation => approval (and adheres to the monotonic principle).

Reply:
For the generic review, why couldn't these same semantics (omission of `reviewer` implies check signer) be achieved with a general predicate that makes `reviewer` optional? 

Two comments on the VCS case: First, why wouldn't we ever want to record a negative review? What if we need forensic evidence of this review or to prove that the threshold number by approved reviewers was not achieved? 

Second, as I understand the monotonic principle, it's the consumer side that should be ensuring their policies follow the monotonic principle; a consumer policy that follows the monotonic principle should be one that says "deny unless results=pass". So by this, an omission of `results` isn't a monotonic way of expressing a positive result. While the producer should enable the consumer to make these types of policy decisions, I don't think the producer should be the one dictating the policy of the consumer. So, I still don't see why all the different use cases can't be achieved with a single code review predicate that makes the appropriate fields optional, allowing for as-rich or as-simple semantics as the consumer may need.

Reply:
> For the generic review, why couldn't these same semantics (omission of reviewer implies check signer) be achieved with a general predicate that makes reviewer optional? This is how SCAI handles the producer field as well, for example.

I think that complicates validating VCS attestations though, as the reviewer is compulsory there.

Does it make more sense to only submit VCS and Crev right now? The generic one is supposed to be more of a catch-all for users to issue reviews but if our use cases largely fall in the VCS / Crev buckets, we can revisit the generic one later.

> VCS case

Also see earlier discussion on this: https://github.com/in-toto/attestation/pull/151#discussion_r1147963407, cc @TomHennen 

Recording a negative review didn't make sense from use cases such as "was this PR reviewed and approved before it was merged". This is in part because we run into the question of when we care about PR reviews. I personally think that only the reviews (approvals) that apply at merge time matter, and that reviews (from the same users often) before the approvals that requested changes and left comments don't affect PR-specific workflows.

Reply:
I've since taken this file out, the PR submits one VCS specific predicate and one for crev.